src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2877,6 +2877,184 @@ export default class EthereumApi implements Api {
+    const totalBlocks = Math.min(
+      Quantity.toNumber(blockCount),
+      newestBlockNumber + 1
+    );
+
+    // Cut out early if no range is given.
+    if (totalBlocks === 0) {
+      return {
+        oldestBlock: Quantity.from(newestBlock).toString(),
+        baseFeePerGas: undefined,
+        gasUsedRatio: null,
+        reward: undefined
+      };
+    }
+
+    // blockCount is inclusive of newestBlock
+    const oldestBlockNumber = newestBlockNumber - (totalBlocks - 1);
+
+    const baseFeePerGas = new Array(totalBlocks);
+    const gasUsedRatio = new Array(totalBlocks);
+    let reward;
+    if (rewardPercentiles.length > 0) {
+      reward = new Array(totalBlocks);
+    }
+
+    let currentBlockNumber = oldestBlockNumber;
+    let currentBlock;
+
+    while (currentBlockNumber <= newestBlockNumber) {
+      currentBlock = await blockchain.blocks.get(
+        Quantity.toBuffer(currentBlockNumber)
+      );
+
+      const currentPosition = currentBlockNumber - oldestBlockNumber;
+
+      baseFeePerGas[currentPosition] =
+        currentBlock.header.baseFeePerGas || Quantity.Zero;
+
+      gasUsedRatio[currentPosition] = Number(
+        `0.${(
+          (currentBlock.header.gasUsed.toBigInt() * PRECISION_BIG_INT) /
+          currentBlock.header.gasLimit.toBigInt()
+        )
+          .toString()
+          .padStart(PAD_PRECISION, "0")}`
+      );
+
+      if (reward) {
+        const transactions = currentBlock.getTransactions();
+
+        // If there are no transactions, all reward percentiles are 0.
+        if (transactions.length === 0) {
+          reward[currentPosition] = rewardPercentiles.map(() => {
+            return Quantity.Zero;
+          });
+        } else {
+          // For all transactions, effectiveGasReward = normalized fee per unit of gas
+          // earned by the miner regardless of transaction type
+          const effectiveRewardAndGasUsed = (
+            await Promise.all(
+              transactions.map(async tx => {
+                const receipt = (
+                  await blockchain.transactionReceipts.get(tx.hash.toBuffer())
+                ).toJSON(currentBlock, tx, blockchain.common);
+
+                const baseFeePerGas = currentBlock.header.baseFeePerGas
+                  ? currentBlock.header.baseFeePerGas.toBigInt()
+                  : 0n;
+
+                let effectiveGasReward;
+                if ("maxPriorityFeePerGas" in tx) {
+                  effectiveGasReward =
+                    tx.maxFeePerGas.toBigInt() - baseFeePerGas;
+                  if (tx.maxPriorityFeePerGas < effectiveGasReward) {
+                    effectiveGasReward = tx.maxPriorityFeePerGas;
+                  }
+                } else {
+                  effectiveGasReward = tx.gasPrice.toBigInt() - baseFeePerGas;
+                }
+
+                return {
+                  effectiveGasReward: Quantity.toBigInt(effectiveGasReward),
```

Comment:
I think if you update the above if / else block to deal with bigint types only, then you won't need to `Quantity.toBigInt`

Reply:
:+1: